### ISSUE TITLE
Move some procesess to GitHost subclasses

### ISSUE THREAD

**Issue by maciekbanas** [r-world-devs/GitStats#367]
This issue somehow reverts to previous GitStats architecture, where `GitHub` and `GitLab` classes existed as subclasses of `GitHost`. This architecture seemed complex, as it introduced an addittional layer to GitStats logic.
Nonetheless, it had one big advantage: managing on `git platform` level which engine to choose when process of getting data is divided between two engines, e.g.:
- pulling commits response via GraphQL, but organizing commits authors logins and names via REST.
- pulling repositories via GraphQL, but repos contributors via REST.

Summing up, although structure of classes grows, the process of pulling data will be constructed through readable "bricks".

**Comment by maciekbanas** [comment]
Some examples to move:
- setting up engines

**Comment by maciekbanas** [comment]
Developed in #383
